testsuite/python/lees_edwards_stress.py

```diff
@@ -0,0 +1,234 @@
+
+    system.non_bonded_inter[0, 0].lennard_jones.set_params(
+                epsilon=1.0, sigma=1.0, cutoff=2.0, shift=0)
+
+    # calculate forces and stress without Lees-Edwards offset
+    le_offset = 0.0
+    system.lees_edwards_offset = le_offset
+    system.integrator.run(steps=0)
+
+    system.part[0].v = [1.0,2.0,3.0]
+    system.part[1].v = [-0.1,-0.2,-0.3]
+
+    pos = system.part[:].pos
+    vel = system.part[:].v
+
+    sim_stress_ideal = system.analysis.stress_tensor()['ideal']
+    sim_stress_bonded = system.analysis.stress_tensor()['bonded']
+    sim_stress_nonbonded = system.analysis.stress_tensor()['non_bonded']
+    sim_stress_total = system.analysis.stress_tensor()['total']
+    sim_pressure_ideal = system.analysis.pressure()['ideal']
+    sim_pressure_bonded = system.analysis.pressure()['bonded']
+    sim_pressure_nonbonded = system.analysis.pressure()['non_bonded']
+    sim_pressure_total = system.analysis.pressure()['total']
+
+    anal_stress_ideal = stress_ideal(vel, box_l)
+    anal_stress_bonded = stress_bonded(pos, box_l)
+    anal_stress_nonbonded = stress_nonbonded(pos, box_l)
+    anal_stress_total = anal_stress_ideal + anal_stress_bonded + anal_stress_nonbonded
+    anal_pressure_ideal = np.einsum('ii', anal_stress_ideal)/3.0
+    anal_pressure_bonded = np.einsum('ii', anal_stress_bonded)/3.0
+    anal_pressure_nonbonded = np.einsum('ii', anal_stress_nonbonded)/3.0
+    anal_pressure_total = anal_pressure_ideal + anal_pressure_bonded + anal_pressure_nonbonded
+
+    print('le_offset = {}'.format(system.lees_edwards_offset))
+
+    print('\nparticle positions')
+    print(pos)
+    print('particle velocities')
+    print(vel)
+
+    print('\nsimulated ideal pressure=')
+    print(sim_pressure_ideal)
+    print('analytical ideal pressure=')
+    print(anal_pressure_ideal)
+
+    print('\nsimulated bonded pressure=')
+    print(sim_pressure_bonded)
+    print('analytical bonded pressure=')
+    print(anal_pressure_bonded)
+
+    print('\nsimulated non-bonded pressure=')
+    print(sim_pressure_nonbonded)
+    print('analytic non-bonded pressure=')
+    print(anal_pressure_nonbonded)
+
+    print('\nsimulated total pressure=')
+    print(sim_pressure_total)
+    print('analytic total pressure=')
+    print(anal_pressure_total)
+
+    print('\nsimulated ideal stress=')
+    print(sim_stress_ideal)
+    print('analytic ideal stress=')
+    print(anal_stress_ideal)
+
+    print('\nsimulated bonded stress=')
+    print(sim_stress_bonded)
+    print('analytic bonded stress=')
+    print(anal_stress_bonded)
+
+    print('\nsimulated non-bonded stress=')
+    print(sim_stress_nonbonded)
+    print('analytic non-bonded stress=')
+    print(anal_stress_nonbonded)
+
+    print('\nsimulated total stress=')
+    print(sim_stress_total)
+    print('analytic total stress=')
+    print(anal_stress_total)
+
+    self.assertTrue(np.max(np.abs(sim_stress_ideal-anal_stress_ideal)) < tol, 'ideal stress does not match analytical result')
```

Comment:
Apparently the current style is to use assertLess and so on, because it produces more helpful error messages than "Fals is not True".